### ISSUE TITLE
[Deployment Issue] Deploy from github using image doesn't work.

### ISSUE THREAD

**Issue by NickNaskida** [caprover/caprover#2058]
What is the problem?
I am following this tutorial in the docs: [Deploying directly from Github | Alternative method](https://caprover.com/docs/ci-cd-integration/deploy-from-github.html#alternative-method-more-efficient). Unfortunetely, my deploy github action workflow fails on the step of "Deploy Image to CapRrover". 

In logs I see it requires a password, which is weird because I am using app token:
![Screenshot 2024-05-12@2x](https://github.com/caprover/caprover/assets/82929931/4479f4b6-fd4f-407e-bfbe-e188d8550585)

I did exactly follow the tutorial, but can't make it to work. All the github secrets are set, github registry is registered in caprover, ...

Here is my workflow file content:
```yml
name: Build & Deploy Staging Backend

env:
  CAPROVER_APP_NAME: app-backend-staging

on:
  push:
    branches: [ "staging" ]

jobs:
  build_and_deploy:
    runs-on: ubuntu-latest

    steps:
    - name: Check out repository
      uses: actions/checkout@v4

    - name: Set up Docker Buildx
      uses: docker/setup-buildx-action@v3

    - name: Login to GitHub Container Registry
      uses: docker/login-action@v3
      with:
        registry: ghcr.io
        username: ${{ github.repository_owner }}
        password: ${{ secrets.GITHUB_TOKEN }}

    - name: Preset Image Name
      run: echo "IMAGE_URL=$(echo ghcr.io/${{ github.repository_owner }}/${{ github.event.repository.name }}:$(echo ${{ github.sha }} | cut -c1-7) | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV

    - name: Build and push Docker Image
      uses: docker/build-push-action@v5
      with:
        context: ./backend
        file: ./backend/Dockerfile
        push: true
        tags: ${{ env.IMAGE_URL }}

    - name: Deploy Image to CapRover
      uses: caprover/deploy-from-github@1.1.2
      with:
        server: '${{ secrets.CAPROVER_SERVER }}'
        app: '${{ env.CAPROVER_APP_NAME }}'
        token: '${{ secrets.CAPROVER_APP_TOKEN }}'
        image: ${{ env.IMAGE_URL }}
```

Steps to reproduce the problem:

Output of the following command on your server:
```
uname -a && lsb_release -a && free -h
```

Output:
```
Linux caprover 5.4.0-177-generic #197-Ubuntu SMP Thu Mar 28 22:45:47 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.6 LTS
Release:	20.04
Codename:	focal
              total        used        free      shared  buff/cache   available
Mem:           11Gi       1.2Gi       6.0Gi        34Mi       4.5Gi        10Gi
Swap:         2.9Gi          0B       2.9Gi
```

**Comment by githubsaturn** [comment]
Are you sure `CAPROVER_APP_TOKEN` is set in your configs. The fact that you're seeing a messaging prompting to input the password means that `CAPROVER_APP_TOKEN` is empty.

To debug:
First, ensure that all of these variables are available at the last step:
```
                  server: ${{ secrets.CAPROVER_HOST }}
                  app: ${{ env.CAPROVER_APP }}
                  token: '${{ secrets.CAPROVER_APP_TOKEN }}'
                  image: '${{ env.FULL_IMAGE_NAME }}'
```

**Comment by NickNaskida** [comment]
So I finally figured it out. My issue was that I was using github environments where I set secrets for this workflow. The thing is, by default, github does run the workflow in the default environment. So after modifying my workflow to use my specified environment according to [this answer](https://stackoverflow.com/a/66526312) on StackOverflow, it finally deployed successfully.

Thank you for your help! Closing the issue
